Add noqa and disable .ansible-lint global exclusions

Miouge1 opened the pull request:
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Remove global ansible-lint exceptions, and replace them with local `# noqa`.

**Additional notes**:
Fixing all the instances of a given rule in one PR is difficult since it occurs in many roles (CNI, container engines, etc...). I suggest to drop the global exception add a bunch of `# noqa` to allow things to be fixed 1 task at a time rather than all or nothing.

I know it's not ideal, to have so many `# noqa`, but global exception is maybe worst?

/cc @MarkusTeufelberger

MarkusTeufelberger commented:
As you see there are only <170 total linter errors for 7 rules, so I would say it is definitely feasible to fix these, especially as some are definitely easier to fix than others.

I would at least add something unique that can be queried since now it is not clear if some ansible-lint rule was violated willingly (e.g. because something needs to be done in a certain way) or just silenced in here to prevent new violations.

This just pushes the work down the road, there is no way around to fixing these eventually and once you enable e.g. 305 and 306 for new contributions you'll have no good examples to point to and will just force this work on potentially less experienced contributors how to write proper tasks that run on all supported (or rather: all tested...) platforms. It would be easier to test some idiom and apply this repository-wide rather than wait until someone sneakily deactivates this in a PR out of frustration again (see #5189 and #5230).

Miouge1 commented:
Thank you for the feedback @MarkusTeufelberger 

I've added a comment in `.ansible-lint` to be more explicit: intentional noqa should have comment, otherwise I think it is safe to assume that they need to be looked into. It is possible to search for unintentional instances with `noqa \d+$`.

To be honest a I used a small Python script to inject the `# noqa` comments based on the output of ansible-lint.

This has been pretty helpful to search the codebase (my editor doesn't have support for ansible-lint rules highlighting, but searching for `# noqa` has been a game changer for me).

In this PR I've added commits to fix E305 and E306 (and a couple of E301). Using this, I've also created another PR for E404 (#6417).

With this work I also identified some playbooks to delete (see #6418).

At this stage, I do not intend to fix ansible-lint errors in `contrib/`, I think it is fair to focus on the core codebase rather than contrib folder.

As it stands in 94394f7 there are mostly E301, E303 and E503 left which I would prefer to address to follow up PRs to reduce the risk of merge conflicts.

floryut commented:
Looks good, CI is happy and lot of works to be done 
/lgtm

Miouge1 commented:
I did a rebase and resolved merge conflicts. Some of the fixes will be split into a subsequent PR to avoid more conflicts.
